topdown/input.go

```diff
@@ -43,21 +43,23 @@ func mergeTermWithValues(exist *ast.Term, pairs [][2]*ast.Term) (*ast.Term, erro
 				result = exist.Copy()
 				init = true
 			}
-
 			if result == nil {
 				result = ast.NewTerm(makeTree(target[1:], pair[1]))
 			} else {
 				node := result
 				done := false
 				for i := 1; i < len(target)-1 && !done; i++ {
-					if child := node.Get(target[i]); child == nil {
-						obj, ok := node.Value.(ast.Object)
-						if !ok {
-							return nil, errConflictingDoc
-						}
+					obj, ok := node.Value.(ast.Object)
+					if !ok {
+						return nil, errConflictingDoc // possible?
```

Comment:
I think the following statement would cause this:

```
> true with input as [] with input.foo as 1
1 error occurred: true with input as [] with input.foo as 1: eval_conflict_error: conflicting documents
```

Basically, if the root is a non-object value we'll error out here. I guess we need to special case that.

Reply:
Yep, indeed. Thank you!